servicetalk-http-api/src/main/java/io/servicetalk/http/api/DefaultHttpExecutionStrategy.java

```diff
@@ -99,17 +104,19 @@ public StreamingHttpService offloadService(final Executor fallback, final Stream
                 // ExecutionStrategy to understand if we need to offload more than we already offloaded:
                 final HttpExecutionStrategy diff = difference(fallback, ctx.executionContext().executionStrategy(),
                         DefaultHttpExecutionStrategy.this);
+
                 // The service should see this ExecutionStrategy inside the ExecutionContext:
                 final HttpServiceContext wrappedCtx =
                         new ExecutionContextOverridingServiceContext(ctx, DefaultHttpExecutionStrategy.this, e);
                 if (diff == null) {
                     return service.handle(wrappedCtx, request, responseFactory);
                 } else {
                     if (diff.isDataReceiveOffloaded()) {
-                        request = request.transformMessageBody(p -> p.publishOn(e));
+                        request = request.transformMessageBody(p ->
+                                p.publishOn(e, DefaultHttpExecutionStrategy::shouldOffload));
```

Comment:
This `shouldOffload` does not verify `isIoThreadSupported()`, consider checking what is in `ctx`